Class/practice2.2/main.cpp

```diff
@@ -0,0 +1,104 @@
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <sstream>
+#include <stdexcept>
+
+using namespace std;
+
+// This function is long enough and it does everything.
+// It is better to split it to few parts - one is related to checks of inputs, another reads data or even better - 
+// we can introduce class for Image and then load and save will be a part of its interface
+// Function to process the input file and generate the output file
+void processFile(const string& inputFileName, const string& outputFileName, const vector<int>& favoriteColor) {
```

Comment:
We have a very long function which is difficult to support and test.
Instead it is better to have Pixel class, Image class and split that logic into methods of the classes